TestPushDiagnostics/publishes_global_diagnostics_after_checking is racy

Saw this in CI:

```
--- FAIL: TestPushDiagnostics (0.00s)
    --- FAIL: TestPushDiagnostics/publishes_global_diagnostics_after_checking (4.30s)
        project_test.go:356: assertion failed: hasGlobalDiag is false: expected a 'Cannot find global' diagnostic on tsconfig.json, got: []
FAIL
FAIL	github.com/microsoft/typescript-go/internal/project	17.435s
```

--- a/internal/project/project_test.go
+++ b/internal/project/project_test.go
@@ -326,6 +326,10 @@ func TestPushDiagnostics(t *testing.T) {
 		// Request semantic diagnostics to trigger checking, which triggers the global type resolvers.
 		ls, err := session.GetLanguageService(projecttestutil.WithRequestID(context.Background()), lsproto.DocumentUri("file:///src/index.ts"))
 		assert.NilError(t, err)
+		// Drain background tasks from DidOpenFile (publishProgramDiagnostics, etc.)
+		// before triggering global diagnostics, to avoid racing with publishGlobalDiagnostics.
+		session.WaitForBackgroundTasks()
+
 		_, err = ls.ProvideDiagnostics(projecttestutil.WithRequestID(context.Background()), lsproto.DocumentUri("file:///src/index.ts"))
 		assert.NilError(t, err)
 		// Enqueue global diagnostics publishing (normally done by the LSP server after each request).

